### ISSUE TITLE
OutputNotUniqueError for different pipelines?

### ISSUE THREAD

**Issue by Enzodtz** [kedro-org/kedro#2590]
## Description
I have two pipelines (`a` and `b`). Both of them have nodes that return `c`. The problem is that, even though they are in different pipelines, kedro is raising `OutputNotUniqueError`. I would like to ask if this makes sense to be raised for different pipelines and not only for the same pipeline. Thanks

## Context
To make it work, I would need to name the output such as `output_a` and `output_b`, which would make the names a lot more complicated then they should, since the context already provides this information.

## Steps to Reproduce
1. Create pipeline a with a node that outputs `c`
2. Create pipeline b with a node that outputs `c`

## Expected Result
I would expect that it could run normally

## Actual Result
It raises `OutputNotUniqueError`

```
OutputNotUniqueError: Output(s) ['c'] are returned by more than one nodes. Node outputs must be unique.       
```

## Your Environment
* kedro, version 0.18.8
* Python 3.9.13
* Windows 10

**Comment by noklam** [comment]
Would you be able to share a minimal example to reproduce this?

**Comment by Enzodtz** [comment]
I wasn't able to reproduce it in a minimal example 🤔. Not sure about what could be causing this

**Comment by noklam** [comment]
Any chance you may have different node pointing to the same output?

**Comment by Enzodtz** [comment]
Yes, I have two nodes with same output, but they are in different pipelines. However, if I do the same on a blank project, the error doesn't happen. Maybe there's some settings on that?
